src/BSeries.jl

```diff
@@ -1453,4 +1456,379 @@ include("latexify.jl")
     include("precompile.jl")
 end
 
+
+
+"""
+        is_energy_preserving(rka::RungeKuttaMethod,order)::Bool
+
+This function checks whether a Runge-Kutta method `rk` is 
+energy-preserving for a given `order`.
+"""
+function is_energy_preserving(rk::RungeKuttaMethod,order)
+#generate bseries 
+    series = bseries(rk, order)
+    #pass it to the main function 'is_energy_preserving'
+    is_energy_preserving(series)
+end
+
+"""
+    is_energy_preserving(map_series)::Bool
+This function checks whether the B-series `series` of a time integration
+method is energy-preserving for Hamiltonian systems - up to the
+[`order`](@ref) of the `series`.
+
+# References
+This code is based on the Theorem 2 of
+- Elena Celledoni, Robert I. McLachlan, Brynjulf Owren, and G. R. W. Quispel. 
+  "Energy-preserving integrators and the structure of B-series."
+  Foundations of Computational Mathematics 10 (2010): 673-693.
+  [DOI: 10.1007/s10208-010-9073-1](https://link.springer.com/article/10.1007/s10208-010-9073-1)
+ """
+function is_energy_preserving(map_series)
+    flow_series = modified_equation(map_series)
+    #check if the first element is map or flow.
```

Comment:
I think you meant to add an "assert" here to check that the input is indeed a map.

Reply:
Implemented